index.js

```diff
@@ -19,114 +18,205 @@ const TranslationReducer = require('./lib/broccoli/translation-reducer');
 
 module.exports = {
   name: 'ember-intl',
-  addonOptions: null,
   isLocalizationFramework: true,
 
-  included() {
+  /** @private **/
+  _options: null,
+
+  /** @private **/
+  _isHost: false,
+
+  /** @private **/
+  _translationTree: null,
+
+  /** @private **/
+  _hostsLocales: null,
+
+  included(app) {
     this._super.included.apply(this, arguments);
 
-    let app = this.app = this._findHost();
-    this.addonOptions = this.intlConfig(app.env);
+    let host = (this.app = this._findHost());
+    this._isHost = app === host;
 
-    let inputPath = this.addonOptions.inputPath || 'translations';
-    this.hasTranslationDir = existsSync(path.join(app.project.root, inputPath));
-    this.projectLocales = this.findLocales();
+    if (this._isHost) {
+      this._options = this.intlConfig(host.env);
+      this._hostsLocales = this.findLocales();
+      this._translationTree = this.getTranslationTree();
+    }
+  },
 
-    let projectTranslations = new WatchedDir(inputPath);
+  /*
+   * Move addon translations into a folder scoped by the addon name
+   * so that the application translations to not collide with addons.
+   *
+   * @method moveNode
+   * @param {node} broccoli tree/node/directory
+   * @param {addon} addon model
+   * @private
+   */
+  moveNode(node, { name }) {
+    return funnel(node, { destDir: name, srcDir: '.' });
+  },
 
-    let addonTranslations = this.findIntlAddons().map(function(addon) {
-      return funnel(addon.path, {
-        srcDir: addon.translationPath,
-        destDir: `__addon__${addon.name}`
-      });
-    }, this);
+  /*
+   * @method getTranslationTree
+   * @private
+   */
+  getTranslationTree() {
+    let nodes = [];
+    let projectTranslationPath = path.join(this.app.project.root, this._options.inputPath);
+    this.processAddons(this.project.addons, nodes);
+
+    if (existsSync(projectTranslationPath)) {
+      let projectTranslationTree = this.treeGenerator(projectTranslationPath);
+      nodes.push(projectTranslationTree);
+    }
+
+    return mergeTrees(nodes, { overwrite: true });
+  },
 
-    this.translationTree = this.mergeTranslationTrees(projectTranslations, addonTranslations);
+  /*
+   * @method processAddons
+   * @param {array} array of addon models
+   * @param {array} array of broccoli node/tree/directories
+   * @private
+   */
+  processAddons(addons, nodes) {
+    addons.forEach(addon => this.processAddon(addon, nodes));
+  },
+
+  /*
+  * @method processAddon
+  * @param {object} addon model
+  * @param {array} array of broccoli node/tree/directories
+  * @private
+  */
+  processAddon(addon, nodes) {
```

Comment:
Ripped from ember-cli-fastboot.  They are the only addon I know that introduces an addon hook and it looked like a good implementation for what I need.